### PR TITLE
mark previous templates/plugins deprecated

### PR DESCRIPTION
**Pull request by pkwarren**
Update to mark references to previous templates/plugin deprecated in the protobuf APIs. Update the buf.gen.yaml example shown in `buf generate --help` to use the `plugin` key and change references to point to managed plugins.

### CODE REVIEW

File: proto/buf/alpha/registry/v1alpha1/authz.proto [hunk 1]
```diff
@@ -84,7 +94,9 @@ service AuthzService {
   rpc UserCanManagePluginContributors(UserCanManagePluginContributorsRequest) returns (UserCanManagePluginContributorsResponse);
   // UserCanManageTemplateContributors returns whether the user is authorized to manage
   // any contributors to the template and the list of roles they can manage.
-  rpc UserCanManageTemplateContributors(UserCanManageTemplateContributorsRequest) returns (UserCanManageTemplateContributorsResponse);
+  rpc UserCanManageTemplateContributors(UserCanManageTemplateContributorsRequest) returns (UserCanManageTemplateContributorsResponse) {
```
Comment (mfridman) [inline proto/buf/alpha/registry/v1alpha1/authz.proto:97]: Why this one, and not the one above like `UserCanManagePluginContributors` ?

Comment (mfridman) [reply proto/buf/alpha/registry/v1alpha1/authz.proto:97]: Sorry should be more clear, what's the criteria for deprecating the rpc?

Comment (pkwarren) [reply proto/buf/alpha/registry/v1alpha1/authz.proto:97]: I'll deprecate that one too. I was trying to target all of the remote generation alpha RPCs.

File: README.md [hunk 2]
```diff
@@ -78,7 +78,7 @@ While we intend to incrementally improve on the `buf` CLI and the [BSR](#the-buf
 
 ## The Buf Schema Registry
 
-The [Buf Schema Registry][bsr] (BSR) is a SaaS platform for managing your Protobuf APIs. It provides a centralized registry and a single source of truth for all of your Protobuf assets, including not just your `.proto` files but also [plugins] and code generation [templates]. Although the BSR provides an intuitive browser UI, `buf` enables you to perform most BSR-related tasks from the command line, such as [pushing] Protobuf sources to the registry and managing [users] and [repositories]. The BSR is currently in [**beta**][bsr_post].
+The [Buf Schema Registry][bsr] (BSR) is a SaaS platform for managing your Protobuf APIs. It provides a centralized registry and a single source of truth for all of your Protobuf assets. Although the BSR provides an intuitive browser UI, `buf` enables you to perform most BSR-related tasks from the command line, such as [pushing] Protobuf sources to the registry and managing [users] and [repositories]. The BSR is currently in [**beta**][bsr_post].
```
Comment (mfridman) [inline README.md:81]: Feels like we should have a link to https://buf.build/plugins ?

File: .github/workflows/buf.yaml [hunk 3]
```diff
@@ -9,19 +9,19 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: bufbuild/buf-setup-action@v1.9.0
-        if: success()
```
Comment (mfridman) [inline .github/workflows/buf.yaml:12]: > A default status check of success() is applied unless you include one of these functions.

https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions

File: .github/workflows/buf.yaml [hunk 4]
```diff
@@ -9,19 +9,19 @@ jobs:
...
+        with:
+          github_token: ${{ github.token }}
```
Comment (mfridman) [inline .github/workflows/buf.yaml:13]: Solves rate limit issues, great!

**Review by mfridman — APPROVED** [review]

**Pull request merged**